fsociety.py

```diff
@@ -686,9 +714,18 @@ def __init__(self):
             self.install()
         clearScr()
         print(self.doorkLogo)
-        target = raw_input("   Enter a Target: ")
-        self.run(target)
-        response = raw_input(continuePrompt)
+        target = raw_input("   Enter a Target: ").split(' ')[0]
+        test_target = ''
+        if target[0:4] == 'http':
+            test_target = target
+        else:
+            test_target = 'http://'+target
+        try:
+            urllib2.urlopen(test_target)
```

Comment:
Same above: the `urllib2.urlopen` checks if the url is valid, but a specially crafted input can still be dangerous and lead to `arbitrary command injection`.

Mik